app/controllers/assignments_controller.rb

```diff
@@ -8,35 +14,45 @@ def create
 
       if shift_assignment_service.perform
         flash[:success] = shift_assignment_service.success
+
       else
         flash[:danger] = shift_assignment_service.errors
       end
     end
 
-    redirect_to home_path
+    redirect_to assignments_new_path
   end
 
   # rubocop:disable Metrics/AbcSize
   # :reek:TooManyStatements
   # :reek:NilCheck
   # :reek:DuplicateMethodCall
   def show
-    if !params.has_key?(:range1) && !params.has_key?(:range2)
-      flash[:danger] = 'Missing date input!'
-      redirect_to home_path
-    end
-
-    @range = date_range(params[:range1], params[:range2])
+    @range = if !params.has_key?(:range1) && !params.has_key?(:range2)
```

Comment:
what if only 1 of the 2 params is missing?

Reply:
Fixed